glyphs-reader/src/font.rs

```diff
@@ -685,23 +712,32 @@ impl RawFont {
 
         // in each font master setup the parallel array
         for master in self.font_master.iter_mut() {
-            let mut metric_dicts = Vec::new();
+            if Some(OrderedFloat(0.0)) == master.cap_height {
+                master.cap_height = None;
+            }
+            if Some(OrderedFloat(0.0)) == master.x_height {
```

Comment:
why special case these two cap_height and x_height? I agree 0.0 is arguably a good value for them, but why assume they are None?

Reply:
glyphsLib defaults to these values when they are omitted

https://github.com/googlefonts/glyphsLib/blob/da1aae7ac1b4c9f3d9e740be74ae2cc59a52dc6f/Lib/glyphsLib/classes.py#L1612-L1613
But to be sure, it might be good to see what Glyphs.app python scripting API returns when loading a font that does not have those set.

Reply:
If I create a new font from Glyphs 3 and immediately save it, I can see that the x-height and cap height metricValues are set respectively to 500 and 700 by default.
If I deliberately remove those two metrics properties from the master tab and save, I can see they disappear from the .glyphs file as expected. However if I open the Macro window and do

```python
print(Glyphs.font.masters[0].xHeight)
```
... I just get garbage that looks like random memory?! (`9223372036854775808`).

Oh... I tried to set the two values to 0 and then saved the file, and now I see what you mean, there's empty {} dict in their places in the metricValues array. It does look like the metrics values are omitted when both `pos` and `over` are zero. It's not just those two that you singlet out here, but all of them. I tried to set to (0,0) the ascender or the descender and they also get written out as empty {} in the serialized .glyphs file.

Reply:
> v3 appears to omit them where v2 sets 0.

so yeah, in v3 metricValues, if either pos or over or both are not present in each metricValue, they are assumed to be 0.0.

in GSMetricValue constructor, in fact, they default to 0:
https://github.com/googlefonts/glyphsLib/blob/da1aae7ac1b4c9f3d9e740be74ae2cc59a52dc6f/Lib/glyphsLib/classes.py#L1442

Reply:
I modified it to not special case those.